### PR TITLE
SH-12: Add language changing button

### PR DESCRIPTION
**Pull request by WeszeRita**
## JIRA ticket link:
https://shoperz.atlassian.net/browse/SH-12

### CODE REVIEW

File: src/app/app.component.spec.ts
Comment (krucsodaniel) [inline src/app/app.component.spec.ts:1]: We don't have test files across our application yet, please remove the spec.ts files for now.

File: src/app/app.module.ts [hunk 1]
```diff
@@ -31,6 +33,8 @@ const routes: Routes = [
 @NgModule({
   declarations: [
     AppComponent,
+    AboutUsComponent,
+    CartComponent,
```
Comment (krucsodaniel) [inline src/app/app.module.ts]: why is the CartComponent in the App module's declarations array?

File: src/app/pages/about-us/about-us.module.ts [hunk 2]
```diff
@@ -18,6 +18,5 @@ const routes: Routes = [
     RouterModule.forChild(routes),
     SvgIconsModule,
   ],
-  declarations: [AboutUsComponent],
```
Comment (krucsodaniel) [inline src/app/pages/about-us/about-us.module.ts:21]: the component shouldn't belong to the AboutUs module's declarations?

File: src/app/pages/cart/cart.module.ts [hunk 3]
```diff
@@ -16,9 +16,12 @@ import { TranslateModule } from '@ngx-translate/core';
     TranslateModule,
   ],
   declarations: [
-    CartComponent,
```
Comment (krucsodaniel) [inline src/app/pages/cart/cart.module.ts]: why the CartComponent was deleted from its module?

File: src/app/pages/products/components/filters/filter-card/filter-card.component.html [hunk 4]
```diff
@@ -13,5 +13,5 @@
       [checked]="isChecked(option.id)"
       (change)="onChange($event)"/>
```
Comment (krucsodaniel) [inline src/app/pages/products/components/filters/filter-card/filter-card.component.html:15]: extra space

File: src/app/pages/products/components/filters/filter-card/filter-card.component.ts [hunk 5]
```diff
@@ -70,4 +70,16 @@ export class FilterCardComponent implements OnInit {
 
     this.control.setValue(this.selectedOptions);
   }
+
+  buildTranslationKey(relativeKey: string): string {
```
Comment (krucsodaniel) [inline src/app/pages/products/components/filters/filter-card/filter-card.component.ts]: add the method a new name please, like buildTranslationKeyForHeading

File: src/shared/components/header/header.component.html [hunk 6]
```diff
@@ -1,7 +1,5 @@
-<header class="flex flex-col md:flex-row justify-center md:justify-start items-center md:items-start text-center md:text-start w-full bg-blue-600 text-blue-100 py-0 px-44">
-  <a class="mx-2.5 my-5">{{ buildTranslationKey('home') | translate }}</a>
+<app-top-header></app-top-header>
 
-  <a class="mx-2.5 my-5" [routerLink]="Route.products" (click)="resetFiltering()">{{ buildTranslationKey('products') | translate }}</a>
+<app-middle-header class="navbar"></app-middle-header>
```
Comment (krucsodaniel) [inline src/shared/components/header/header.component.html]: we can remove the class attribute if we dont use it

File: src/shared/components/header/header.component.html [hunk 7]
```diff
@@ -1,7 +1,5 @@
...
 
-  <a class="mx-2.5 my-5">{{ buildTranslationKey('faq') | translate }}</a>
-</header>
+<app-sub-header class="header"></app-sub-header>
```
Comment (krucsodaniel) [inline src/shared/components/header/header.component.html]: we can remove the class attribute if we dont use it

File: src/shared/components/top-header/top-header.component.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<div class="flex flex-row px-40 py-2 bg-grey-100 text-grey-600">
+    <svg-icon [src]="'localization-icon'"></svg-icon>
```
Comment (krucsodaniel) [inline src/shared/components/top-header/top-header.component.html]: add empty line after svg please

File: src/shared/components/top-header/top-header.component.html [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+  <select (change)="onChange($event)">
+    <option disabled selected>Language</option>
```
Comment (krucsodaniel) [inline src/shared/components/top-header/top-header.component.html]: just for styling purposes, could you try adding a bit different, for example more pale color to the selected disabled option, for better UI

File: src/shared/enums/langs.ts
Comment (krucsodaniel) [inline src/shared/enums/langs.ts:1]: the names of ENUMs are given in singular, for example export enum Language {...}, don't forget to change the file name as well

File: src/shared/components/top-header/top-header.component.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export class TopHeaderComponent {
+  readonly dropdownOptions = [
+    { value: Langs.english, label: 'English'},
+    { value: Langs.hungarian, label: 'Magyar'},
+    { value: Langs.german, label: 'Deutsch'},
+  ];
+
+  constructor(private translateService: TranslateService) {}
+
+  onChange(event: any){
```
Comment (krucsodaniel) [inline src/shared/components/top-header/top-header.component.ts]: add Event type to parameter

**Review by krucsodaniel — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts (hunk 1 above)
Comment (WeszeRita) [reply src/app/app.module.ts]: there is no good reason for that, I moved it cart module :)

File: src/app/pages/cart/cart.module.ts [hunk 11]
```diff
@@ -20,5 +20,9 @@ import { TranslateModule } from '@ngx-translate/core';
     CartItemComponent,
     CartSummaryComponent,
   ],
+  exports: [
+    CartItemComponent,
+    CartSummaryComponent
```
Comment (krucsodaniel) [inline src/app/pages/cart/cart.module.ts]: add comma after component

**Review by krucsodaniel — CHANGES_REQUESTED** [review]

File: src/shared/components/header/header.component.html (hunk 7 above)
Comment (WeszeRita) [reply src/shared/components/header/header.component.html]: we use it, the unit test needs it

File: src/app/app.component.spec.ts
Comment (WeszeRita) [reply src/app/app.component.spec.ts:1]: I tried, it is not possible :D

File: src/app/pages/cart/cart.module.ts [hunk 12]
```diff
@@ -20,5 +20,9 @@ import { TranslateModule } from '@ngx-translate/core';
     CartItemComponent,
     CartSummaryComponent,
   ],
+  exports: [
+    CartItemComponent,
+    CartSummaryComponent,
+  ]
```
Comment (bodnya29179) [inline src/app/pages/cart/cart.module.ts]:
```suggestion
  ],
```

File: src/app/pages/products/components/filters/filter-card/filter-card.component.html [hunk 13]
```diff
@@ -1,17 +1,16 @@
...
     *ngFor="let option of options"
     class="list-none text-grey-700 m-1.5"
     [ngClass]="multiselect ? 'cursor-pointer' : 'cursor-pointer hover:font-extrabold'"
     [class.checked-label]="isChecked(option.id)">
     <input
+      (change)="onChange($event)"
+      [checked]="isChecked(option.id)"
+      [name]="multiselect ? null : 'radio'"
       [ngClass]="multiselect ? 'cursor-pointer' : 'appearance-none'"
       [type]="multiselect ? 'checkbox' : 'radio'"
```
Comment (bodnya29179) [inline src/app/pages/products/components/filters/filter-card/filter-card.component.html]: change the order as it was

File: src/shared/components/top-header/top-header.component.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  readonly dropdownOptions = [
+    { value: Language.english, label: 'English'},
+    { value: Language.hungarian, label: 'Magyar'},
+    { value: Language.german, label: 'Deutsch'},
+  ];
+
+  constructor(private translateService: TranslateService) {}
+
+  onChange(event: Event){
+    const element = event.target as HTMLSelectElement;
```
Comment (bodnya29179) [inline src/shared/components/top-header/top-header.component.ts]: add space & return type

File: src/shared/components/top-header/top-header.component.ts [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    this.translateService.use(element.value);
```
Comment (bodnya29179) [inline src/shared/components/top-header/top-header.component.ts]: use the value property directly in const

**Review by bodnya29179 — CHANGES_REQUESTED** [review]

**Review by krucsodaniel — APPROVED** [review]
Nice job, approved! :)

**Review by bodnya29179 — APPROVED** [review]

**Pull request merged**